op-devstack/dsl/supervisor.go

```diff
@@ -161,18 +161,22 @@ func (s *Supervisor) WaitForL2HeadToAdvanceTo(chainID eth.ChainID, lvl types.Saf
 			"chain", chainID, "label", lvl, "current", chStatus.Number, "target", blockID.Number)
 		if chStatus.Number < blockID.Number {
 			return fmt.Errorf("expected %s head to advance to blockID: %v", lvl, blockID)
-		} else if chStatus == blockID {
-			return nil // success
 		} else if chStatus.Number == blockID.Number && chStatus.Hash != blockID.Hash {
 			err := fmt.Errorf("supervisor %s head with blockID %v for chainID %s does not match target blockID: %v", lvl, chStatus, chainID, blockID)
 			cancel(err)
 			return err
-		} else { // if chStatus.Number > blockID.Number
-			err := fmt.Errorf("supervisor %s head with blockID %v for chainID %s already advanced past target blockID: %v", lvl, chStatus, chainID, blockID)
-			cancel(err)
-			return err
 		}
+		return nil
```

Comment:
Even if we reached here and no error, there may be a chance that supervisor never reached `blockID` since we do not check the hash. Supervisor may already passed the `blockID.number` with different hash. We may log as a warning when this happens

Reply:
This is not possible to verify in the supervisor, since it doesn't have getBlockByHashForChainID() or getBlockByNumberForChainID() APIs. In any case the only place this is called immediately after an independently checked upgrade in `[testSupervisorAnchorBlock()](https://github.com/ethereum-optimism/optimism/blob/develop/op-acceptance-tests/tests/interop/upgrade/post_test.go#L95)` within `[TestPostInteropUpgradeComprehensive()](https://github.com/ethereum-optimism/optimism/blob/develop/op-acceptance-tests/tests/interop/upgrade/post_test.go#L42)`.